framework/dora_api/infrastructure/middleware.py

```diff
@@ -38,18 +39,42 @@ async def deserialise_web_request():
     _RequestEndpoint = request.endpoint.split(".")[-1]
     if _RequestEndpoint in REQUEST_BODYS_BY_ENDPOINT:
         _RequestData: dict = request.get_json()
+        _DeserialisedRequestData : dict = dict()
 
         for _AttributeName, _AttributeType in get_type_hints(REQUEST_BODYS_BY_ENDPOINT[_RequestEndpoint]).items():
-            _Data = _RequestData[_AttributeName]
+            _Data = _RequestData.get(_AttributeName)
+
+            _AttributeTypeOrigin = get_origin(_AttributeType)
+            if _AttributeTypeOrigin is not None and issubclass(_AttributeTypeOrigin, AttributeChangeTracker):
+                __deserialise_attribute_change_tracker(_AttributeName, _RequestData, _DeserialisedRequestData)
+                continue
 
             if _AttributeType is bytes:
                 _Data = b64decode(_Data)
 
-            _RequestData[_AttributeName] = _AttributeType(_Data) if _Data else None
+            _DeserialisedRequestData[_AttributeName] = _AttributeType(_Data) if _DeserialisedRequestData else None
 
-        _DeserialisedRequest = REQUEST_BODYS_BY_ENDPOINT[_RequestEndpoint](**_RequestData)
+        _DeserialisedRequest = REQUEST_BODYS_BY_ENDPOINT[_RequestEndpoint](**_DeserialisedRequestData)
         setattr(request, "request_body", _DeserialisedRequest)
 
+def __deserialise_attribute_change_tracker(attribute_name: str, src_data: dict, dest_data: dict):
+    '''
+        Overrides, or sets the value of request_data[attribute_name] to the deserialised AttributeChangeTracker value.
+
+        Args:
+            attribute_name (str): The name of the attribute which requires deserialisation.
+            request_data (dict): The data which may contain the attribute_name as a key.
+    '''
+    #TODO: Update comment
+    if attribute_name not in src_data.keys():
+        dest_data[attribute_name] = AttributeChangeTracker()
```

Comment:
Rather than sending dest in and assigning, might be clearer and less code to just return the attribute change tracker and assign the returned value of this method where you call it

Reply:
good idea, refactored